packages/taquito/src/subscribe/observable-subscription.ts

```diff
@@ -24,8 +29,13 @@ export class ObservableSubscription<T> implements Subscription<T> {
             this.call(this.closeListeners);
           }
         ),
-        this.shouldRetry ? retryWhen(errors => errors.pipe(take(retries))) : tap(),
-        catchError(() => timer(delayInMS))
+        this.shouldRetry ? operatorFunction() : tap(),
+        catchError(() => {
+          delayInMS = errorFlag ? delayInMS*2 : delayInMS;
+          delayInMS = delayInMS < maxDelayInMS ? delayInMS : maxDelayInMS;
+          errorFlag = true;
+          return timer(delayInMS);
```

Comment:
By accepting an operator as a param, I was thinking that we should let the user decide his own retry logic, I don't understand why we need to include this logic in the catch block. Can you explain, please?